docs/cunumeric/source/user/practices.rst

```diff
@@ -0,0 +1,411 @@
+of the array rather than a view like numpy, so this deviation can cause
+differences in results, as shown in the example below. This additional copy
+can also make it run slower, so we recommend using it as sparingly as possible.
+
+.. code-block:: python
+
+    import cunumeric as np
+
+    x = np.ones((3,4))
+    y = x.reshape((12,))
+
+    y[0] = 42
+
+    assert x[0,0] == 42 # succeeds in NumPy, fails in cuNumeric
+
+Stack
+.....
+
+There is a performance penalty to stacking arrays using
+`hstack <https://numpy.org/doc/stable/reference/generated/numpy.hstack.html#numpy-hstack>`_
+or
+`vstack <https://numpy.org/doc/stable/reference/generated/numpy.vstack.html#numpy-vstack>`_
+because they incur additional copies of data in our implementation.
+
+I/O Routines
+~~~~~~~~~~~~
+
+As of 23.07, we recommend using `h5py <https://github.com/h5py/h5py>`_ to perform I/O.
+
+Guidelines on designing cuNumeric applications
+----------------------------------------------
+
+Use Output argument
+~~~~~~~~~~~~~~~~~~~
+
+Whenever possible, use the ``out`` parameter in the APIs, to avoid allocating an
+intermediate array in our implementation.
+
+.. code-block:: python
+
+    import cunumeric as np
+
+    # Acceptable
+    x = x + y
+    y = x - y
+    x = x * y
+
+    # Recommended for better performance
+    np.add(x, y, out=x)
+    np.subtract(x, y, out=y)
+    np.multiply(x, y, out=x)
+
+
+Vectorize
+~~~~~~~~~
+
+Functions with conditionals that operate on scalars might make array-based
+operations less straightforward. The general recommendation in such cases is to
+apply the three step process mentioned here where we evaluate the conditional
+and then apply it for both the ``if`` and ``else`` statements. Here is an
+example of what approaches might or might not work. The first and second
+options have ``if`` and ``else`` clauses written out as separate array-based
+operations while the third option (using the API ``where``) includes them both
+in one API.
+
+.. code-block:: python
+
+    # Works with scalars but not NumPy arrays
+    def bar(x):
+        if x < 0:
+            return x + 1
+        else:
+            return x + 2
+
+    # Not Recommended for arrays
+    x = np.array(...)
+    y = bar(x) # doesn't work
+
+    # Recommended (1): Use array-based operations
+    cond = x < 0
+    x[cond] += 1
+    x[~cond] += 2
+
+    # Recommended (2): Use array-based operations
+    cond = x < 0
+    np.add(x, 1, where=cond, out=x)
+    np.add(x, 2, where=~cond, out=x)
+
+    # Recommended (3): Use array-based operations
+    cond = x < 0
+    x = np.where(cond, x + 1, x + 2)
+
+
+Merge Tasks
+~~~~~~~~~~~
+
+It is recommended that tasks (e.g., a Python operation like ``z = x + y``,
+will be a task) be large enough to execute for at least a millisecond to
+mitigate the runtime overheads associated with launching a task. One way to
+make the tasks execute for longer is to merge them when possible. This is
```

Comment:
"longer" instead of "for longer"?

Reply:
"for longer" seems common in my experience, FWIW I actually think it seems better in this specific context. I'll leave this as-is unless there is strong objection